src/components/campaignguide/ScenarioView.tsx

```diff
@@ -201,8 +219,9 @@ function ScenarioView({ componentId, campaignId, showLinkedScenario, processedSc
         });
       }
     }
+
     queueNarration(queue);
```

Comment:
So every time a scenario step changes (i.e. they make a decision/input of any kind), this will trigger a `queueNarration` event.  Does that handle deduplication?

Reply:
Trying to think if there's a better way to handle this... and maybe there isn't.  Regardless, I think the name should probably be something like, `setScenarioNarration` rather than `queueNarration`, since it is really an 'all at once' call and not something iterative (queue makes me think its adding the new elements to the queue?)

Reply:
> Does that handle deduplication?
It does

Reply:
> `setScenarioNarration`

Maybe `setNarrationQueue`? and change `playNarration` to `playNarrationTrack`